app/models/armor.rb

```diff
@@ -104,7 +104,7 @@ def set_enchantments
       enchantables_enchantments.find_or_create_by!(
         enchantment_id: model.enchantment_id,
         strength: model.strength,
-      )
+      ) {|new_model| new_model.added_automatically = true }
```

Comment:
This is a little-known feature of `find_or_create_by!` where a block can be passed to add attributes to a created model that are not used to find an existing model.